include/host/wasi_crypto/symmetric/aeads/cipher.h

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
I'm not sure: most helper class indicates a specific pattern(return `invalid_operation` for invalid operation) . If we need to create new algorithm(eg. blake3 for hash function) , we can simple add default behavior by inherit helper class. But if we don't need create new, merge is a good option to avoid overdesigning